src/utils/thread_pool.hpp

```diff
@@ -23,6 +23,16 @@
 
 namespace memgraph::utils {
 
+template <typename Func>
+struct CopyMovableFunctionWrapper {
+  CopyMovableFunctionWrapper(Func &&func) : func_{std::make_shared<Func>(std::move(func))} {}
+
+  void operator()() { (*func_)(); }
+
+ private:
+  std::shared_ptr<Func> func_;
+};
```

Comment:
How come you didn't end up using Gareth's movable function wrapper

Reply:
Gareth's function wrapper is a move-only function.
This means we would need to update the ThreadPool to use it and that is out of scope of this PR.